solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -275,10 +281,12 @@ public void onStart() {
                 // https://github.com/fullstorydev/lucene-solr/commit/445508adb4a
                 long delayed = System.nanoTime() - requestSubmitTimeNanos;
                 if (delayed > DELAY_WARN_THRESHOLD) {
+                  long delay = TimeUnit.MILLISECONDS.convert(delayed, TimeUnit.NANOSECONDS);
                   log.info(
-                      "Remote shard request to {} delayed by {} milliseconds",
-                      req.servers,
-                      TimeUnit.MILLISECONDS.convert(delayed, TimeUnit.NANOSECONDS));
+                      "Remote shard request to {} delayed by {} milliseconds", req.servers, delay);
+                  if (delayedRequestlistener != null) {
+                    delayedRequestlistener.accept(delay);
```

Comment:
Ideally, there shouldn't be any delay. I was bit thinking about it. 
1. Would be good to reduce DELAY_WARN_THRESHOLD to say 200 millisecond
2. Maybe we want metrics for `how many requests got delayed` ?

Reply:
> Would be good to reduce DELAY_WARN_THRESHOLD to say 200 millisecond

done
> Maybe we want metrics for how many requests got delayed ?

doesn't the count show it

Reply:
I'm inclined to agree with @hiteshk25 in preferring histogram here.

I think the only possible reason to prefer a meter to a histogram is if the default histogram impl introduces too much overhead (it involves locking iiuc, whereas default meter impl does not). The downside of a meter is that what we care most about is the absolute latency of the highest outliers, which p99 or whatever with a histogram would give you a window into, but meter (given sufficient number of lower-latency requests to offset) would not, or not in as transparent a way.

I was initially thinking that it makes sense to respect something like `DELAY_WARN_THRESHOLD` for logging only, but I now think that if we're going the histogram route, we don't actually care about the fast requests. Maybe there's a lot of requests that are fast -- great, whatever. What we _most_ want to know is: among the _bad_ requests, just how bad _are_ they? (this would also mitigate any possible performance impact of metrics collection, by simply not collecting metrics for the common low-latency case).

I do think we should still have logging as well, since it captures useful information about the requests involved. I think we should set it to `WARN` level though, now that we basically expect that this should not happen anymore, and if it does it's a problem. Should we have the threshold level(s?) for WARN logging and metric updating be configurable? Could even have default (no threshold set) turn off metrics collection and/or logging by default.

Reply:
Yeah, histogram is better